api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+      parameters:
+        - name: name
+          in: path
+          description: Name of the EventType to update.
+          type: string
+          required: true
+        - name: event-type
+          in: body
+          description: EventType to be updated.
+          schema:
+            type: object
+            $ref: '#/definitions/EventType'
+          required: true
+        - name: X-Flow-Id
+          in: header
+          description: |
+            The flow id of the request, which is written into the logs and passed to called services. Helpful
+            for operational troubleshooting and log analysis.
+          type: string
+          format: flow-id
+      responses:
+        '200':
+          description: Ok
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: EventType not found.
+          schema:
+              $ref: '#/definitions/Problem'
+        '422':
+          description: Unprocessable Entity
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Server error
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /event-types/{name}/events:
+    post:
+      tags:
+        - stream-api
+      description: |
+        Publishes one or more events of this EventType.
+
+        Reception of Events will always respect the configuration of its EventType w.r.t. validation,
+        enrichment and partitioning. The steps performed on reception of incoming message are:
+
+        1. Every validation rule specified in the `EventType` will be checked in order against the incoming
+        Events. Validation rules are evaluated in the order they are defined and the Event is **rejected**
+        in the first case of failure with a status of 400 Bad request. If the offending validation rule
+        provides information about the violation it will be included in the Problem response.
+        If the `EventType` defines schema validation it will be performed at this moment.
+        **TBD behaviour in case of violation within a stream of events**)
+
+        1. Once the validation succeeded, the content of the Event is updated according to the enrichment
+        rules in the order the rules are defined in the `EventType`.
+        No preexisting value might be changed (even if added by an enrichment rule). Violations on this will
+        force the immediate **rejection** of the Event with a 409 Conflict status. The invalid overwrite
+        attempt will be included in the respose's Problem object.
+
+        1. The incoming Event's relative ordering is evaluated according to the rule on the
+        `EventType`. Failure to evaluate the rule will **reject** the Event.
+        **Restriction**: POSTing Events based on its `EventType` requires it to have a
+        `PartitioningStrategy` defined. Violating this condition will produce a 405 Not allowed.
```

Comment:
What I mean is that you have to implement the logic to actually sent different product-ids in parallel but not the same. This is reimplementing the partitioning decision in the producer.
I agree that batching will help here though, so that should work for most requirements.